addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/handler/KNXBasicThingHandler.java

```diff
@@ -0,0 +1,682 @@
+    }
+
+    private void withKNXType(Channel channel, ChannelFunction function) {
+        try {
+            KNXChannelType selector = getKNXChannelType(channel);
+            if (selector != null) {
+                Configuration channelConfiguration = channel.getConfiguration();
+                if (channelConfiguration != null) {
+                    function.apply(selector, channelConfiguration);
+                } else {
+                    logger.warn("The configuration of channel {} is empty", channel.getUID());
+                }
+            } else {
+                logger.warn("The KNX channel type {} is not implemented", channel.getChannelTypeUID().getId());
+            }
+        } catch (KNXException e) {
+            logger.error("An error occurred on channel {}: {}", channel.getUID(), e.getMessage());
+            if (logger.isDebugEnabled()) {
+                logger.error("", e);
+            }
+        }
+    }
+
+    private void forAllChannels(ChannelFunction function) {
+        for (Channel channel : getThing().getChannels()) {
+            withKNXType(channel, function);
+        }
+    }
+
+    @Override
+    public void channelLinked(ChannelUID channelUID) {
+        withKNXType(channelUID, (selector, configuration) -> {
+            Boolean mustRead = (Boolean) configuration.get(READ);
+            BigDecimal readInterval = (BigDecimal) configuration.get(INTERVAL);
+            for (GroupAddress address : selector.getReadAddresses(configuration)) {
+                if (mustRead || readInterval.intValue() > 0) {
+                    scheduleReadJob(address, selector.getDPT(address, configuration), true, BigDecimal.ZERO);
+                }
+            }
+        });
+    }
+
+    private void scheduleReadJobs() {
+        cancelReadFutures();
+
+        for (Channel channel : getThing().getChannels()) {
+            if (isLinked(channel.getUID().getId())) {
+                withKNXType(channel, (selector, channelConfiguration) -> {
+                    Boolean mustRead = (Boolean) channelConfiguration.get(READ);
+                    BigDecimal readInterval = (BigDecimal) channelConfiguration.get(INTERVAL);
+                    for (GroupAddress address : selector.getReadAddresses(channelConfiguration)) {
+                        scheduleReadJob(address, selector.getDPT(address, channelConfiguration), mustRead,
+                                readInterval);
+                    }
+                });
+            }
+        }
+    }
+
+    private void scheduleReadJob(GroupAddress groupAddress, String dpt, boolean immediate, BigDecimal readInterval) {
+        if (knxScheduler == null) {
+            return;
+        }
+
+        boolean recurring = readInterval != null && readInterval.intValue() > 0;
+
+        if (immediate) {
+            knxScheduler.schedule(new ReadRunnable(groupAddress, dpt), 0, TimeUnit.SECONDS);
+        }
+
+        if (recurring && readInterval != null) {
```

Comment:
well spotted! I just put it there in order to shut up the IDE, as Eclipse seems to be not a believer of boolean logic.